File: 435-non-overlapping-intervals/non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end());
        vector<vector<int>> ans;
        vector<int> temp;
        int count = 0;
        int previous_e = intervals[0][1];
        int previous_s = intervals[0][0];
        for(int i=1;i<intervals.size();i++)
        {
            cout<<i<<" "<<previous_e<<endl;
            if(intervals[i][0]<previous_e)
            {
                count++;
                previous_e = min(previous_e,intervals[i][1]);

            }
            else{
                previous_e = intervals[i][1];
                previous_s = intervals[i][0];
                
            }
        }
        return count;
        
    }
};

File: 435-non-overlapping-intervals/non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals,(int[] a, int[] b)->{return a[0]-b[0];});
        int s = intervals[0][0];
        int e = intervals[0][1];
        int count = 0;
        for(int i=1;i<intervals.length;i++)
        {
             if(e>intervals[i][0])
             {
                e = Math.min(e,intervals[i][1]);
                count++;
             }
             else {
                s = intervals[i][0];
                e = intervals[i][1];
             }
        }
        return count;
    }
}